pkg/controller/nodeipam/ipam/cloud_cidr_allocator.go

```diff
@@ -253,43 +255,60 @@ func (ca *cloudCIDRAllocator) updateCIDRAllocation(nodeName string) error {
 		return fmt.Errorf("node %s doesn't have providerID", nodeName)
 	}
 
-	cidrs, err := ca.cloud.AliasRangesByProviderID(node.Spec.ProviderID)
+	ipv6DualStack := utilfeature.DefaultFeatureGate.Enabled(features.IPv6DualStack)
```

Comment:
We shouldn't depend on this. If the AliasRangesByProviderID function returns 2 ranges then we should accept both. This is because in the future, the feature gate may be on by default but the cluster can still just be a single stack cluster

Reply:
Got it. Removed using ipv6DualStack FeatureGate flag and using len(podCIDRs) to check is the same IPs already exist in spec.PodCIDR(s)

Reply:
I added len(podCIDRs) == 1 check to handle the case where we have IPV4 only CIDR and it is populated in node.Spec.PodCIDR (and may not be populated in node.Spec.PodCIDRs[0] ? ).
Ideally node.Spec.PodCIDRs[0] should also be populated. In case of upgrades (like IPV4 to Dual Stack), it might not be populated.